index.html

```diff
@@ -35,58 +41,150 @@
+      </table>
+
+      <section>
+        <h3><dfn>constructor()</dfn> method</h3>
+
+        <aside class="example">
+          In order to communicate via TCP a socket connection must be requested first. The socket object constructor
+          allowes the site to specify the necessary parameters which control how data is transmitted and received.
+
+          <pre class="js">
+            const socket = new TCPSocket(/*remoteAddress=*/, /*remotePort=*/, /*options=*/);
+          </pre>
+
+          The developer should wait for the connection to be established before accessing the {{TCPSocket/connection}} attribute. 
+
+          <pre class="js">
+            const { readable, writable } = await socket.connection;
+          </pre>
+
+          Once {{TCPSocket/connection}} has resolved, the {{TCPSocketConnection/readable}} and {{TCPSocketConnection/writable}}
+          can be accessed to get the {{ReadableStream}} and {{WritableStream}} instances for receiving data from and sending data to the connected socket.
+        </aside>
+
+        The {{TCPSocket/constructor()}} steps are:
+
+        <ol>
+          <li>If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+        not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets=]", throw a
+        "{{NotAllowedError}}" {{DOMException}} and return `null`.
+          <li>If only one of |options|["{{TCPSocketOptions/localAddress}}"] or |options|["{{TCPSocketOptions/localPort}}"] is specified, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is not, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is not specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAliveDelay}}"] is less than one second, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/sendBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/receiveBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>Perform the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to open the TCP socket using the given |remoteAddress| and |remotePort| and the 
+                connection parameters (or their defaults) specified in |options|.
+              <li>
+                If this fails for any reason:
+                <ol>
+                  <li>[=Reject=] the {{TCPSocket/[[connectionPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
+                  <li>Detach {{TCPSocket/[[closedPromise]]}}.
```

Comment:
Is there something you can link to for "detach"?

I think you actually want to reject [[closedPromise]] with the same exception as [[connectionPromise]].

Reply:
I tried to look it up and didn't find anything :(
My reasoning here is that the socket cannot be closed if it hasn't been fully open, therefore it's more appropriate to leave this promise in a hanged state and forget about it rather than reject it.

Reply:
I think leaving the promise unsettled will make it easy for developers to write bugs. The way I look at it `opened` and `closed` are like events but they are only ever fired once. If opening fails then by definition the socket is closed.